.circleci/config.yml

```diff
@@ -64,6 +67,58 @@ jobs:
             open_pr:true \
             automatic_release:<< pipeline.parameters.automatic >>
 
+  detekt:
+    executor: jdk17
+    steps:
+      - checkout
+      - android/restore-gradle-cache
+      - run:
+          name: Run Detekt on commonMain
+          command: ./gradlew detektCommonMain
+      - android/save-gradle-cache
+      - run:
+          name: Collect Detekt HTML reports
+          command: mkdir -p artifacts/detekt && mv build/reports/detekt/*.html artifacts/detekt/
+      - store_artifacts:
+          path: artifacts/detekt
+          destination: detekt
+
+  validate-binary-compatibility:
+    executor: xcode15
```

Comment:
I enabled native ABI checking, so it also checks the "public api" from an iOS perspective, instead of only from a JVM perspective. This requires a Mac host. See [this note](https://github.com/Kotlin/binary-compatibility-validator?tab=readme-ov-file#klib-abi-dump-generation-and-validation-on-linux-and-windows-hosts) in the docs.

Reply:
Ah nice!